crypto/bio/bss_mem.c

```diff
@@ -296,13 +314,18 @@ static long mem_ctrl(BIO *b, int cmd, long num, void *ptr)
         if (ptr != NULL) {
             pptr = (char **)ptr;
             *pptr = (char *)(bm->data);
+            b->flags |= BIO_FLAGS_DATA_FETCH;
         }
         break;
     case BIO_C_SET_BUF_MEM:
         mem_buf_free(b);
         b->shutdown = (int)num;
         bbm->buf = ptr;
         *bbm->readp = *bbm->buf;
+        if (!b->shutdown)
```

Comment:
What if the user changes `b->shutdown` by calling `BIO_CTRL_SET_CLOSE` *after* calling this?